File: test/mercury/arith.m
:- module arith.
:- interface.
:- import_module int.

% Pure arithmetic for bridge bisimulation testing.
% Expected: compute(2) = 2 * (3 + 4) + 5 * 6 = 44
:- func compute(int) = int.

:- implementation.
compute(X) = X * (3 + 4) + 5 * 6.

File: test/haskell/Arith.hs
module Arith where

-- | Pure arithmetic — no IO, no typeclasses beyond Num Int.
-- Used for bridge bisimulation testing.
-- Expected result: 2 * (3 + 4) + 5 * 6 = 14 + 30 = 44
compute :: Int -> Int
compute x = x * (3 + 4) + 5 * 6
